optaplanner-core/src/main/java/org/optaplanner/core/api/solver/SolverFactory.java

```diff
@@ -36,9 +37,11 @@
  * Creates {@link Solver} instances.
  * Most applications only need one SolverFactory.
  * <p>
- * To build an instance, use {@link #createFromXmlResource(String)}.
- * To change the configuration programmatically, create a {@link SolverConfig}
- * and use {@link #create(SolverConfig)}.
+ * To create an SolverFactory, use {@link #createFromXmlResource(String)}.
```

Comment:
Minor: should be "a SolverFactory" instead of "an SolverFactory"

Reply:
Thanks for the review, Ron :)
What do you think of the naming? SolverManager, SolverJob, ...
and the example's code size?

Reply:
Overall, I think the class naming is pretty good and makes sense. Only alternative I could think of is something like SolverPoolManager or something that indicates its a "pool" of solver threads. A few minor thoughts:
- Could use a bit more class javadoc on SolverJob
- I'm not sure about the `solveBatch` method name as it kind of makes me think you are sending in a 'batch' of problems to solve but that doesn't appear to be the case
- I don't fully understand why there are 3 different `solveBatch` methods with varying parameters and whether they are all useful (I see they just chain together so not a huge deal) but reading through I don't really understand when I would use one vs another
- I do agree that `updateScore` seems a little out of place so might be interesting in considering alternatives to avoid exposing that there

Reply:
We went to rename solveBatch to solve(), more overloaded methods, clearer javadocs to explain which to use when.
We also renamed solveObserving() to solveAndListen()